### ISSUE TITLE
Help with updating a Cask

### ISSUE THREAD

**Issue by miccal** [caskroom/homebrew-cask#17060]
I have been trying to update a Cask via the command line, but I am not sure what I am doing wrong and would like some help please. I have successfully created a cask following the fantastic instructions in `CONTRIBUTING.md`, but trying to edit a Cask is casing me problems.

For example, I would like to edit `android-file-transfer.rb` (beyond what `cask-repair` does). I do
```bash
$ github_user='<my-github-username>'
$ cd "$(brew --repository)"/Library/Taps/caskroom/homebrew-cask
$ git remote add "$github_user" "https://github.com/$github_user/homebrew-cask"
```
then
```bash
$ brew cask edit android-file-transfer.rb
```
then edit the file in Vim, then `:wq`, then
```bash
$ cd "$(brew --repository)"/Library/Taps/caskroom/homebrew-cask
$ git status
```
which shows that `android-file-transfer.rb` has been modified. Then
```bash
$ git checkout -b android-file-transfer
Switched to a new branch 'android-file-transfer'
$ git add Casks/android-file-transfer.rb
```
I check `git diff --cached` to make sure my changes are accurate, and then `git commit -v`. Finally I
```bash
$ github_user='<my-github-username>'
$ git push "$github_user" android-file-transfer
```
where I enter my username and password, then
```bash
$ cd "$(brew --repository)"/Library/Taps/caskroom/homebrew-cask
$ git checkout master
```

Now, when I go to the [`homebrew-cask` GitHub repository](https://github.com/caskroom/homebrew-cask), my branch is not there. When I check my own fork, a branch has been created but does not have my commits, it shows the previous commit (from someone else).

Worse, running `brew cask list` produces an error about the file `android-file-transfer.rb` having a problem, and checking the file shows it is the one I edited. To fix this I run
```bash
$ cd "$(brew --repository)"/Library/Taps/caskroom/homebrew-cask
$ git checkout master
```
which reports I am a commit ahead, and have to run `git reset --hard origin/master` to fix `brew cask list`.

Am I missing a step? I am relatively new to git.

Thanks.

**Comment by adityadalal924** [comment]
@miccal Thank you for the detailed report.

Can you look through this (abbreviated) guide to contributing? It only lists the bash commands, in a concise format - CONTRIBUTING might be a bit too wordy to get the picture clearly across

https://gist.github.com/adityadalal924/9e6dd7630f62ec737685

**Comment by miccal** [comment]
Thanks for your help @adityadalal924.

Output of `git remote -v`:
```bash
USER	https://github.com/USER/homebrew-cask (fetch)
USER	https://github.com/USER/homebrew-cask (push)
origin	https://github.com/caskroom/homebrew-cask (fetch)
origin	https://github.com/caskroom/homebrew-cask (push)
```
Output of `git status`:
```bash
On branch master
Your branch is up-to-date with 'origin/master'.
nothing to commit, working directory clean
```
Output of `git checkout -b android-file-transfer`:
```bash
fatal: A branch named 'android-file-transfer' already exists.
```
I assume because I created it previously (although I though `git reset --hard` would have deleted this?). Instead I do `git checkout android-file-transfer`, with output:
```bash
Switched to branch 'android-file-transfer'
Your branch is up-to-date with 'USER/android-file-transfer'
```
but then I do `vim android-file-transfer.rb`, which created a new file instead of editing the existing Cask.

Thanks again!

**Comment by adityadalal924** [comment]
`checkout -b` creates a new branch and checks it out. To fully clean things up, delete the branch from the github web ui and:

``
git branch -D android-file-transfer #deletes branch locally
git reset --hard origin/master # resets to caskroom/master
git push "USER" master --force
```

and then follow the instructions from the gist.

**Comment by miccal** [comment]
OK, I have done what you suggested and tried again. This time `git checkout -b android-file-transfer` returned `Switched to a new branch 'android-file-transfer'`, but then I do `vim android-file-transfer.rb`, which again creates a new file instead of editing the existing Cask.

I am not sure if this is relevant, but the output of `git branch --all` before I created the new branch is:

```bash
* master
  remotes/USER/delete-alternate
  remotes/USER/delete-changelog
  remotes/USER/delete-faq
  remotes/USER/installer-unpack
  remotes/USER/master
  remotes/origin/HEAD -> origin/master
  remotes/origin/master
```

Thanks again for taking the time to help.

**Comment by miccal** [comment]
I think I have figured it out and I have created a pull request.

Thanks @adityadalal924 for all your help.

**Comment by miccal** [comment]
Just one more thing to learn!

My original pull request has a syntax error, so I edited it and created another commit. When trying to squash the commits with `git rebase -i origin/master` I had the `pick` and `squash` around the wrong way, and received the error `Cannot 'squash' without a previous commit`. 

So I tried again, and received

```bash
It seems that there is already a rebase-merge directory, and
I wonder if you are in the middle of another rebase.  If that is the
case, please try
	git rebase (--continue | --abort | --skip)
If that is not the case, please
	rm -fr "/usr/local/Library/Taps/caskroom/homebrew-cask/.git/rebase-merge"
and run me again.  I am stopping in case you still have something
valuable there.
```

So I removed as suggested, and now my commits are gone. When I `git rebase -i origin/master` I get `noop` on the first line, and my commits are gone from `git log`.

Can I fix this?
